backend/src/main/java/com/github/valfink/backend/fooditem/FoodItemService.java

```diff
@@ -85,7 +96,37 @@ public FoodItemDTOResponse addFoodItem(FoodItemDTORequest foodItemDTORequest, Mu
 
     public FoodItemDTOResponse getFoodItemById(String id) {
         FoodItem foodItem = foodItemRepository.findById(id)
-                .orElseThrow(NoSuchElementException::new);
+                .orElseThrow(() -> new NoSuchElementException("The task with the id " + id + " doesn't exist."));
+
+        return foodItemDTOResponseFromFoodItem(foodItem);
+    }
+
+    public FoodItemDTOResponse updateFoodItemById(String id, FoodItemDTORequest foodItemDTORequest, MultipartFile photo, Principal principal) {
+        String userId = mongoUserService.getMongoUserDTOResponseByUsername(principal.getName()).id();
+        FoodItemDTOResponse oldFoodItem = getFoodItemById(id);
+
+        if (!oldFoodItem.donator().id().equals(userId)) {
+            throw new SecurityException("You may only edit you own items!");
+        }
+        throwExceptionIfFoodItemDTORequestIsNotValid(foodItemDTORequest);
+
+        String photoUri;
+        if (oldFoodItem.photoUri() == null || oldFoodItem.photoUri().isBlank()) {
+            photoUri = uploadPhotoIfPresentAndReturnUriOrNull(photo);
+        } else {
+            photoUri = oldFoodItem.photoUri();
+        }
+
+        FoodItem foodItem = foodItemRepository.save(new FoodItem(
+                id,
+                userId,
+                foodItemDTORequest.title(),
+                photoUri,
+                foodItemDTORequest.location(),
+                foodItemDTORequest.pickupUntil(),
+                foodItemDTORequest.consumeUntil(),
+                foodItemDTORequest.description()
+        ));
 
         return foodItemDTOResponseFromFoodItem(foodItem);
     }
```

Comment:
The code in the class is very good. If more coding is added here in the future, you could create another helper class